server/webapp/WEB-INF/rails.new/app/controllers/admin/tasks_controller.rb

```diff
@@ -67,17 +72,25 @@ def initialize params, user, security_service, task, pluggable_task_service
         end
 
         def subject(job)
+          if @task.is_a?(com.thoughtworks.go.config.AbstractFetchTask)
+            return com.thoughtworks.go.config.FetchTaskAdapter.new(@task)
+          end
           @task
         end
 
         def update(job)
-          job.addTask(@task)
+          if @task.is_a?(com.thoughtworks.go.config.FetchTaskAdapter)
+            job.addTask(@task.getAppropriateTask)
+          else
+            job.addTask(@task)
+          end
           @pluggable_task_service.validate(@task) if @task.instance_of? com.thoughtworks.go.config.pluggabletask.PluggableTask
           @pluggable_task_service.validate(@task.cancelTask()) if (!@task.cancelTask().nil?) && (@task.cancelTask().instance_of? com.thoughtworks.go.config.pluggabletask.PluggableTask)
         end
       end.new(params, current_user.getUsername(), security_service, @task, pluggable_task_service), create_failure_handler, {:controller => '/admin/tasks', :current_tab => params[:current_tab]}) do
         assert_load :job, @node
         assert_load :task, @subject
+        assert_load :artifact_ids, ["foo", "bar"].to_json
```

Comment:
This needs to populated all artifact ids that can be used, right? I think this is being used for autocomplete. But I might be wrong.

Reply:
That line is not needed.